borg/testsuite/repository.py

```diff
@@ -216,18 +216,18 @@ def segments_in_repository():
         self.repository.commit()
 
         self.repository.append_only = False
-        assert segments_in_repository() == 1
+        assert segments_in_repository() == 2
         self.repository.put(b'00000000000000000000000000000000', b'foo')
         self.repository.commit()
         # normal: compact squashes the data together, only one segment
-        assert segments_in_repository() == 1
+        assert segments_in_repository() == 4
```

Comment:
so the final state is like this?:

segment file: content
---------------------------
(0: deleted)
1: commit
2: put 0 foo
3: commit
4: commit

that looks strange.